### PR TITLE
[js/webgpu] allow setting env.webgpu.adapter

### PR DESCRIPTION
**Pull request by fs-eire**
### Description
Allow user to set `env.webgpu.adapter` before creating the first inference session.

Feature request: https://github.com/microsoft/onnxruntime/pull/19857#issuecomment-1999984753

@xenova

### CODE REVIEW

**Review by guschmue — DISMISSED** [review]

**Review by xenova — APPROVED** [review]
This looks great! Thanks a ton!

File: js/web/lib/wasm/wasm-core-impl.ts [hunk 1]
```diff
@@ -93,18 +93,31 @@ export const initEp = async(env: Env, epName: string): Promise<void> => {
...
+        }
+        const forceFallbackAdapter = env.webgpu.forceFallbackAdapter;
+        if (forceFallbackAdapter !== undefined && typeof forceFallbackAdapter !== 'boolean') {
+          throw new Error(`Invalid forceFallbackAdapter setting: "${forceFallbackAdapter}"`);
+        }
+        adapter = await navigator.gpu.requestAdapter({powerPreference, forceFallbackAdapter});
+        if (!adapter) {
+          throw new Error(
+              'Failed to get GPU adapter. ' +
+              'You may need to enable flag "--enable-unsafe-webgpu" if you are using Chrome.');
```
Comment (guschmue) [inline js/web/lib/wasm/wasm-core-impl.ts:113]: is -enable-unsafe-webgpu still a thing?

**Review by guschmue — APPROVED** [review]

File: js/web/lib/wasm/wasm-core-impl.ts (hunk 1 above)
Comment (fs-eire) [reply js/web/lib/wasm/wasm-core-impl.ts:113]: We can later update this message to a link on onnxruntime.ai, after the troubleshooting page is done.

**Comment by xenova** [comment]
I just ran into a use-case where, in addition to `adapter`, it would be useful to set the `device` too (requesting/creating a device with [higher limits](https://developer.mozilla.org/en-US/docs/Web/API/GPUSupportedLimits#examples)). Is this something that can be added?

**Comment by fs-eire** [comment]
> I just ran into a use-case where, in addition to `adapter`, it would be useful to set the `device` too (requesting/creating a device with [higher limits](https://developer.mozilla.org/en-US/docs/Web/API/GPUSupportedLimits#examples)). Is this something that can be added?

No problem. However, what is ort-web's expected behavior if user set both `adaptor` and `device` while the `device` is not created from the `adapter`? If there is no API to check, we probably just need to assume they never make this mistake.

**Pull request merged**